sympy/solvers/bivariate.py

```diff
@@ -199,17 +199,40 @@ def _solve_lambert(f, symbol, gens):
     nrhs, lhs = f.as_independent(symbol, as_Add=True)
     rhs = -nrhs
 
+    even_degrees = [i for i in degree_list(lhs) if i%2 == 0]
```

Comment:
`degree_list(lhs, symbol)` will be better, we are only concerned with the `symbol`'s degree.

Reply:
Okay It will be included in next commit.

Reply:
`degree_list(lhs, symbol)` raises this error `sympy.polys.polyerrors.PolynomialError: exp(2*_X0*I/x) contains an element of the set of generators.` I dont know how to fix this.

Reply:
Since `degree_list` takes `*gens` as the arguments it throws a polynomialerror if the generator is contained in something else. It expects the generator to be completely free. Even `y**x` for `gen` `x` throws exception.

Also I see another concern using `degree_list`, it returns the degree but we require all power of the concerned variable to be returned like consider `f = x**2 + x**3 + y**2`, we require `(2, 3)` to be returned and not just `(3, )` which is the case with `degree_list` (There is a difference between `degree` and `power`).
Maybe we require a new routine to fulfil our task. The API could be something like `power_list(f, *symbols)`. This will return all powers of the desired symbols.

Reply:
Ahh I got your point!!. Okay I will be creating `power_list` in a separate PR. Can you please tell me where I should put `power_list`. I mean I saw the working of `degree_list`, it was very complex. I want to know how should it work and where should it be placed.

Reply:
@Yathartha22 @aktech I had created a function to return powers of symbols present in expression in #17043 . Please review and let get it merged so that we could fast up the process.

Reply:
You said "we are also considering here equations a*x**even + b* f(containing odd powers of x)" and agreed that "then you should do the replacement (symbol -> t) only when len == 1 above". That's why I made my suggestion. (And to make work, what I showed, `exeq` in `sift` should be `exeq.args`.

Reply:
> You said "we are also considering here equations ax**even + b f(containing odd powers of x)"

Appologies if you felt that way. 
> (And to make work, what I showed, exeq in sift should be exeq.args.)

This also I tried was not giving the things that we expected 
```python
from sympy.utilities.iterables import sift
from sympy.core import Mul
print(lhs, "lhs")
lhs = lhs.expand()
odd, even = sift(lhs.args, lambda t: any(
        i.is_Pow and i.base == symbol and i.exp % 2 == 0
        for i in Mul.make_args(t)), binary=True)
print(even, "even")
# for various test cases
4*2**(2*p + 3) - 2*p lhs
[-2*p, 32*2**(2*p)] even

-x**2 + exp(x*log(2)) lhs
[exp(x*log(2))] even

-_x*log(2) + 2*log(-_x) lhs
[2*log(_x), -_x*log(2), 2*I*pi] even
```

Reply:
> Apologies if you felt that way.

No offense. I am just saying that what you said and what you are saying are not agreeing.

Instead of `odd, even` it should be `even, odd`:
```python
symbol=x
for lhs in (4*2**(2*p + 3) - 2*p,
 -x**2 + exp(x*log(2)),
 -x*log(2) + 2*log(-x)):
 lhs = lhs.expand()
 print(lhs)
 even, odd = sift(lhs.args, lambda t: any(
        i.is_Pow and i.base == symbol and i.exp % 2 == 0
        for i in Mul.make_args(t)), binary=True)
 print (even)

output:
```console
32*2**(2*p) - 2*p
[]
-x**2 + exp(x*log(2))
[-x**2]
-x*log(2) + 2*log(-x)
[]
```

Reply:
Ohh okay okay that I think can be worked upon. Let me work it locally I will get back to you.